crates/ide-assists/src/handlers/sort_items.rs

```diff
@@ -121,24 +122,33 @@ trait AddRewrite {
 impl AddRewrite for Assists {
     fn add_rewrite<T: AstNode>(
         &mut self,
+        ctx: &AssistContext<'_>,
         label: &str,
         old: Vec<T>,
         new: Vec<T>,
         target: TextRange,
     ) -> Option<()> {
+        let node = old.first().unwrap().syntax().parent().unwrap();
```

Comment:
Try instead modifying `add_rewrite` to accept `target: SyntaxNode`

Reply:
Updated.
But, `Assists::add` actually accepts `TextRange` as a parameter, so I had to use `target.text_range()` for that. Let me know if this works better.

Reply:
Awesome, and now you can call `make_editor` with `target` instead!

Also, you can probably take `target: &SyntaxNode` instead to avoid the clones at the callers

Reply:
Right. updated